In [9]:
import pandas as pd
import numpy as np

from sklearn.pipeline import make_pipeline, Pipeline
from sklearn.linear_model import LogisticRegression
from sklearn.ensemble import BaggingClassifier
from sklearn.model_selection import train_test_split
from sklearn.preprocessing import MinMaxScaler
from sklearn.metrics import recall_score

In [24]:
heart=pd.read_csv('framingham.csv')
heart.dropna(inplace=True)
X=heart[['age', 'currentSmoker', 'totChol', 'BMI', 'heartRate']] 
y=heart['TenYearCHD']
X_train,X_test,y_train,y_test=train_test_split(X,y, test_size=0.2, stratify=y)
# Logistic Regression
logit_md=make_pipeline(MinMaxScaler(), 
                       LogisticRegression()).fit(X_train, y_train)
logit_pred=logit_md.predict_proba(X_test)[:, 1]
logit_label=np.where(logit_pred<0.1, 0, 1)
print(f"The recall of the logistic regression model is {recall_score(y_test, logit_label)}")
# Bagging Logistic
bag_md=BaggingClassifier(estimator=make_pipeline(MinMaxScaler(), 
                                                 LogisticRegression()),           
                                    n_estimators=40).fit(X_train, y_train)
bag_pred=bag_md.predict_proba(X_test)[:,1]
bag_label=np.where(bag_pred<.1, 0, 1)
print(f"The recall of the bagged logistic regression model is {recall_score(y_test, bag_label)}")

The recall of the logistic regression model is 0.8928571428571429
The recall of the bagged logistic regression model is 0.8660714285714286


Based on these results:...

The recall of the logistic regression model is 0.8928571428571429
The recall of the bagged logistic regression model is 0.8660714285714286

...I would use the logistic regression model without bagging.